tag/key.go

```diff
@@ -15,9 +15,19 @@
 
 package tag
 
+import (
+	"sync"
+)
+
+var keys = struct {
```

Comment:
this seems unnecessary, why doesn't NewKey just check the string and return Key{name}?